templates/hosts_ini.tpl

```diff
@@ -1,14 +1,14 @@
 [redpanda]
 %{ for i, ip in redpanda_public_ips ~}
-${ ip } ansible_user=${ ssh_user } ansible_become=True private_ip=${redpanda_private_ips[i]}%{ if rack[i] != null } rack=${rack[i]}%{ endif }%{ if tiered_storage_enabled } tiered_storage_bucket_name=${tiered_storage_bucket_name} aws_region=${aws_region}%{ endif }
+${ ip } ansible_user=${ ssh_user } ansible_become=True private_ip=${redpanda_private_ips[i]}%{ if rack[i] != null } rack=${rack[i]}%{ endif }%{ if tiered_storage_enabled } tiered_storage_bucket_name=${tiered_storage_bucket_name} cloud_storage_region=${cloud_storage_region}%{ endif }
 %{ endfor ~}
 %{ if enable_monitoring }
 
 [monitor]
 ${ monitor_public_ip } ansible_user=${ ssh_user } ansible_become=True private_ip=${ monitor_private_ip }
 %{ endif }
-%{ if client_count > 0 }
 
+%{ if length(client_public_ips) > 0 }
```

Comment:
this will end up a noop if the user adds clients with no public ips, right? or is the TF only capable of building clients with public front ends right now?

Reply:
My understanding is that this would be an array of nulls if the clients don't have public ip addresses. Aiming for a noop if there are no clients.

Reply:
In AWS we have: 
```
output "client" {
  value = {
    for instance in aws_instance.client :
    instance.public_ip => instance.private_ip
  }
}
```
In Azure:
```
output "client_public_ips" {
  value = "${azurerm_public_ip.redpanda_client.*.ip_address}"
}
```
And GCP:
```
google_compute_instance.client.*.network_interface.0.access_config.0.nat_ip
```

I think this is okay, especially given the next line:
```
%{ for i, ip in client_public_ips ~}
${ ip } ansible_user=${ ssh_user } ansible_become=True private_ip=${client_private_ips[i]} id=${i}
%{ endfor ~}
```

Reply:
I've just re-tested this on all three. We actually don't give the user an option to configure these without a public IP address. I'm sure they could change it to do so, but also bear in mind that public doesn't have to mean _public_, it could still be a subnet based IP address.
I think we're good here.